### PR TITLE
Added links to all the metric files

### LINKED ISSUE #2437: Add an explanation on how metrics work with engine
## 📚 Documentation

It is unclear for newcomers what does the code below from docs (for example, Accuracy)
```python
metric = Accuracy()
metric.attach(default_evaluator, "accuracy")
y_true = torch.Tensor([1, 0, 1, 1, 0, 1])
y_pred = torch.Tensor([1, 0, 1, 0, 1, 1])
state = default_evaluator.run([[y_pred, y_true]])
print(state.metrics["accuracy"])
```
More precisely, it is unclear that the metric is computed online and at each iteration some internal variables are updated and on epoch completed metric value is computed.

For each metric docs, let's add a link on the explanation that we can add explicitly to https://pytorch.org/ignite/metrics.html

### PR DESCRIPTION
**Pull request by sayantan1410**
Fixes #2437
Opened this PR as an extension of this PR #2468, I by mistake lost track of that branch. Apology or the inconvenience. Extremely sorry.
@vfdev-5 @DevPranjal Kindly let me know if the changes are alright for you or I have made some mistake.  

Check list:
- [X] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@sayantan1410 thanks for the PR !

Actually, I think we decided a bit too fast about including this text to defaults.rst as now it is associated with metrics, however, we are also using it for certain handlers:

https://deploy-preview-2474--pytorch-ignite-preview.netlify.app/generated/ignite.handlers.param_scheduler.create_lr_scheduler_with_warmup.html#ignite.handlers.param_scheduler.create_lr_scheduler_with_warmup

@sayantan1410 @DevPranjal Any ideas how to deal with that ?

**Comment by sayantan1410** [comment]
@vfdev-5 There are two solution in my mind :
1. Either to add the text to all the metric docstring individually, basically the one we rejected earlier. 
2. To create another rst file like default.rst and include it in only the metric docstrings. And remove the previous default.rst from the metric docstrings.

**Comment by vfdev-5** [comment]
In Idea (2) there could be few difficulties like those we had with defaults.rst (we have to copy it into a specific path to be discovered), we have to ensure that python definitions aren't duplicated and we still have to update all docstring for metrics to rename `defaults.rst` -> `default_metric.rst`.

Idea (1) is more explicit but can be a pain to do...

**Comment by sayantan1410** [comment]
@vfdev-5 Not really, I will just have to copy and paste the text to all the metric docstrings !!
Let's wait for @DevPranjal 's thoughts and then we can move forward with this I think.

**Comment by vfdev-5** [comment]
@sayantan1410  I provided a temporary CI fix: https://github.com/pytorch/ignite/pull/2477
Once it is merged please sync your PR with master and we can land it.

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR @sayantan1410

**Comment by sayantan1410** [comment]
@vfdev-5 hey, some checks were unsuccessful that's why it did not merge, can you check once why ??

**Comment by vfdev-5** [comment]
@sayantan1410 yes, there is an issue with new release of openai gym and deprecated env cartpole. we have to fix it.

**Comment by vfdev-5** [comment]
I think I can merge your PR as failures are unrelated.

**Comment by sayantan1410** [comment]
@Thanks for closing the PR !!

**Comment by vfdev-5** [comment]
@sayantan1410 can you please open an issue describing the failed tests and maybe a fix ?

**Comment by sayantan1410** [comment]
@vfdev-5 yeah sure I will do it.
